### PR TITLE
pseudo-sync access directory, not throw exception on absent lazy populating file

### PR DESCRIPTION
**Pull request by magibney**
this must be forward-ported to `fs/branch_9x`

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/storage/AccessDirectory.java [hunk 1]
```diff
@@ -180,11 +180,30 @@ public void rename(String source, String dest) throws IOException {
...
-    // complete, which by definition a lazily reconstituted file is not).
+    ensureOpen();
+
+    for (String name : names) {
+      try {
+        fsync(name); // this covers the normal, actively-writing case.
+      } catch (NoSuchFileException | FileNotFoundException ex) {
+        try {
+          if (activeLazy.get(name) != null
+              || CompressingDirectory.readLengthFromHeader(compressedPath.resolve(name)) > 0) {
```
Comment (hiteshk25) [inline solr/core/src/java/org/apache/solr/storage/AccessDirectory.java:191]: just remind me : we write to persistent and access file parallely, right?

Comment (magibney) [reply solr/core/src/java/org/apache/solr/storage/AccessDirectory.java:191]: Always write to persistent copy first, then access. Sync is parallel, but failure to sync to either access or persistent causes overall sync failure (so when TeeDirectory.sync() returns, it's guaranteed to have sync'd the persistent copy)

Comment (hiteshk25) [reply solr/core/src/java/org/apache/solr/storage/AccessDirectory.java:191]: you mean access copy will be sync'ed eventually?

**Review by hiteshk25 — COMMENTED** [review]
LGTM

**Pull request merged**